simulator/fault_injection.go

```diff
@@ -320,5 +451,80 @@ func (f *FaultInjector) ResetStats() {
 
 	for _, rule := range f.rules {
 		rule.count = 0
+		rule.skipCount = 0
+	}
+}
+
+// parsePropertyFilters parses an array of string filter expressions and consolidates them
+// into a single property.Match map. Only equality matches are supported: "property.path == value"
+// Panics on code-level usage errors:
+//   - Empty filter strings
+//   - Invalid filter format (missing " == " separator)
+//   - Empty property path or expected value
+//   - Multiple filters targeting the same property path (user should combine them into a single pattern)
+func parsePropertyFilters(filterStrs []string) property.Match {
+	match := make(property.Match)
+
+	for _, filterStr := range filterStrs {
+		filterStr = strings.TrimSpace(filterStr)
+		if filterStr == "" {
+			panic("empty filter string in property filter - remove empty entries from filter array")
+		}
+
+		parts := strings.SplitN(filterStr, " == ", 2)
+		if len(parts) != 2 {
+			panic(fmt.Sprintf("invalid property filter format %q - expected format: \"property.path == value\"", filterStr))
+		}
+
+		propertyPath := strings.TrimSpace(parts[0])
+		expectedValue := strings.TrimSpace(parts[1])
+		if propertyPath == "" {
+			panic(fmt.Sprintf("empty property path in filter %q - expected format: \"property.path == value\"", filterStr))
+		}
+		// nil expected values are ok - that's how we expect to match empty properties
+
+		// Convert property path to camelCase (property collector format)
```

Comment:
Why not just take camelCase as input, instead of converting here? We take camelCase as input elsewhere.

Reply:
It's easy to convert and just one less format thing for a user to worry about.
I'm happy to change it to require camelCase as the input form or merge as is. Preference?

It's not blocking yet - my other work is stacked on top of my hickeng branch, but that's some way off merging and this only blocks the merge, not the dev/test.